store/sqlstore/upgrade.go

```diff
@@ -1011,6 +1011,49 @@ func upgradeDatabaseToVersion535(sqlStore *SqlStore) {
 
 	sqlStore.CreateColumnIfNotExists("SidebarCategories", "Collapsed", "tinyint(1)", "boolean", "0")
 
+	sqlStore.CreateColumnIfNotExistsNoDefault("Channels", "TotalMsgCountRoot", "bigint", "bigint")
+	sqlStore.CreateColumnIfNotExistsNoDefault("Channels", "LastRootPostAt", "bigint", "bigint")
+	defer sqlStore.RemoveColumnIfExists("Channels", "LastRootPostAt")
+	ta := time.Now()
+	sqlStore.CreateColumnIfNotExists("ChannelMembers", "MsgCountRoot", "bigint", "bigint", "0")
+	mlog.Info("CreateColumnIfNotExists MsgCountRoot", mlog.String("took", time.Since(ta).String()))
+
+	gt1 := time.Now()
+	c1 := `
+		SELECT Channels.Id channelid, COALESCE(COUNT(*),0) newcount, COALESCE(MAX(Posts.CreateAt), 0) as lastpost
+		FROM Channels
+		LEFT JOIN Posts ON Channels.Id = Posts.ChannelId
+		WHERE Posts.RootId = ''
+		GROUP BY Channels.Id
+	`
+	c2 := "SELECT TotalMsgCountRoot, Id, LastRootPostAt from Channels"
+	uberQ := `
+		WITH q AS (` + c1 + `)
+		UPDATE Channels SET TotalMsgCountRoot = q.newcount, LastRootPostAt=q.lastpost
+		FROM q where q.channelid=Channels.Id;
+
+		WITH q as (` + c2 + `)
+		UPDATE ChannelMembers CM SET MsgCountRoot=TotalMsgCountRoot
+		FROM q WHERE q.id=CM.ChannelId AND LastViewedAt >= q.lastrootpostat;
+	`
+	if sqlStore.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		uberQ = `
+			UPDATE Channels
+			INNER Join (` + c1 + `) as q
+			ON q.channelid=Channels.Id
+			SET TotalMsgCountRoot = q.newcount, LastRootPostAt=q.lastpost;
+
+			UPDATE ChannelMembers CM
+			INNER JOIN (` + c2 + `) as q
+			ON q.id=CM.ChannelId and LastViewedAt >= q.lastrootpostat
+			SET MsgCountRoot=TotalMsgCountRoot
+			`
+	}
+	if _, err := sqlStore.GetMaster().Exec(uberQ); err != nil {
+		mlog.Error("Error fetching Channels table", mlog.Err(err))
+	}
```

Comment:
This is failing on MySQL with:

```go
{"level":"error","ts":1617104190.3134878,"caller":"sqlstore/upgrade.go:1053","msg":"Error fetching Channels table","error":"Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'UPDATE ChannelMembers CM\n\t\t\tINNER JOIN (SELECT TotalMsgCountRoot, Id, LastRootPo' at line 12"}
```

The query looks fine. Maybe it's a formatting issue.

Reply:
On a second look it might actually be cause of the double query. We may need to execute them separately.